larevt/Filters/LArG4ParticleFilter_module.cc

```diff
@@ -27,57 +27,62 @@ namespace filt{
 
     private:
 
-      bool IsInterestingParticle(const art::Ptr<simb::MCParticle> particle, int index);
-      bool PDGCheck(const art::Ptr<simb::MCParticle> particle, int index);
-      bool MinMomentumCheck(const art::Ptr<simb::MCParticle> particle, int index);
-      bool MaxMomentumCheck(const art::Ptr<simb::MCParticle> particle, int index);
-      bool StartInTPCCheck(const art::Ptr<simb::MCParticle> particle, int index);
-      bool StopInTPCCheck(const art::Ptr<simb::MCParticle> particle, int index);
-      bool TPCTrajLengthCheck(const art::Ptr<simb::MCParticle> particle, int index);
+      bool IsInterestingParticle(const art::Ptr<simb::MCParticle> &particle, const unsigned int index);
+      bool PDGCheck(const art::Ptr<simb::MCParticle> &particle, const unsigned int index);
+      bool IsPrimaryCheck(const art::Ptr<simb::MCParticle> &particle, const unsigned int index);
+      bool MinMomentumCheck(const art::Ptr<simb::MCParticle> &particle, const unsigned int index);
+      bool MaxMomentumCheck(const art::Ptr<simb::MCParticle> &particle, const unsigned int index);
+      bool StartInTPCCheck(const art::Ptr<simb::MCParticle> &particle, const unsigned int index);
+      bool StopInTPCCheck(const art::Ptr<simb::MCParticle> &particle, const unsigned int index);
+      bool TPCTrajLengthCheck(const art::Ptr<simb::MCParticle> &particle, const unsigned int index);
```

Comment:
How many of the  member  functions can be `const`-qualified?  For example:

```diff
- bool IsInterestingParticle(const art::Ptr<simb::MCParticle> &particle, const unsigned int index);
+ bool IsInterestingParticle(const art::Ptr<simb::MCParticle> &particle, const unsigned int index) const;
```

Reply:
Good spot.  Literally all of the functions (except for `art::filter`) can be.  I actually thought I'd already done them.  Serves me right for believing that '5 years ago' me knew what I was doing.

Thanks for looking over the PR!